### PR TITLE
Backend Quarkus: Initial Setup of Modern Application

### PR DESCRIPTION
**Pull request by AdlerFleurant**
Created a gradle and maven based quarkus application

Signed-off-by: Adler Fleurant <2609856+AdlerFleurant@users.noreply.github.com>

### CODE REVIEW

File: quarkus/settings.gradle.kts [hunk 1]
```diff
@@ -0,0 +1,12 @@
+rootProject.name = "community-ninja-board"
```
Comment (etsauer) [inline quarkus/settings.gradle.kts:1]: I don't think we need a context root anymore. This app will have a unique domain/route, so we should clean up the URL by serving from ROOT.

Comment (AdlerFleurant) [reply quarkus/settings.gradle.kts:1]: This is a gradle construct equivalent to the maven: `<artifactId>community-ninja-board</artifactId>`

Comment (etsauer) [reply quarkus/settings.gradle.kts:1]: ok great.

Comment (AdlerFleurant) [reply quarkus/settings.gradle.kts:1]: @sabre1041 , it's going to be hard to argue for Maven vs Gradle. I am willing to maintain both a Gradle and Maven until we can have enough data to weight the benefits of 1 vs the other. Sounds good?

**Comment by AdlerFleurant** [comment]
> First and foremost. I'm not cool and still running JDK8 so it fails for me :(
> 
> [Kotlin on Quarkus](https://quarkus.io/guides/kotlin) is marked as Tech Preview. I would think that we would want to produce content that is fully supported now that Quarkus is GA.

@sabre1041 changed it to java only.

File: quarkus/user-service/src/main/java/com/redhat/services/ninja/controller/UserResource.java [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    @Produces(MediaType.APPLICATION_JSON)
+    public RedHatUser findByUid(@PathParam("uid") String uid) {
+        return search("uid", uid).stream().findFirst().orElseThrow(() -> new WebApplicationException(Response.Status.NOT_FOUND));
+    }
+
+    @GET
+    @Path("/{key}/{value}")
+    @Produces(MediaType.APPLICATION_JSON)
+    public List<RedHatUser> find(@PathParam("key") String key, @PathParam("value") String value) {
+        System.out.println("Working");
```
Comment (sabre1041) [inline quarkus/user-service/src/main/java/com/redhat/services/ninja/controller/UserResource.java:30]: Can you remove this debugging line?

Comment (AdlerFleurant) [reply quarkus/user-service/src/main/java/com/redhat/services/ninja/controller/UserResource.java]: My mistake! Corrected!

**Pull request merged**